### PR TITLE
solution

### PR DESCRIPTION
**Pull request by Zelinskyi-Serhii**

### CODE REVIEW

File: src/figuresClasses.ts [hunk 1]
```diff
@@ -1,19 +1,83 @@
...
+enum FigureType {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
 }
 
-export class Triangle {
-
+export interface Figure {
+  color: string;
```
Comment (kateryna-babina) [inline src/figuresClasses.ts:10]: Why type of color string?  Maybe, you should use the type Color that you created in a first line

File: src/figuresClasses.ts [hunk 2]
```diff
@@ -1,19 +1,83 @@
...
+    public b: number,
+    public c: number,
+  ) {
+    if (this.a < 1 || this.b < 1 || this.c < 1) {
+      throw new Error('Wrong value for Triangle');
+    }
+
+    if (this.a + this.b <= this.c) {
+      throw new Error('Wrong value for Triangle');
+    }
```
Comment (kateryna-babina) [inline src/figuresClasses.ts:30]: Are you sure about this? What if we have a = 3, b = 1, c = 1 ? Check all cases :)

**Review by kateryna-babina — CHANGES_REQUESTED** [review]
GJ, but check some issues ;)

File: src/figuresClasses.ts [hunk 3]
```diff
@@ -1,19 +1,87 @@
-export interface Figure {
+type Color = 'red' | 'green' | 'blue';
```
Comment (sergii-nosachenko) [inline src/figuresClasses.ts:1]: The enum fits better for this

File: src/figuresClasses.ts [hunk 4]
```diff
@@ -1,19 +1,87 @@
...
-
+export interface Figure {
+  color: Color;
+  shape: FigureType;
+  getArea(): number;
 }
 
-export class Circle {
+export class Triangle implements Figure {
+  shape = FigureType.Triangle;
```
Comment (sergii-nosachenko) [inline src/figuresClasses.ts:16]:
```suggestion
  readonly shape = FigureType.Triangle;
```
Figures have constant shape

File: src/figuresClasses.ts [hunk 5]
```diff
@@ -1,19 +1,87 @@
...
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
```
Comment (sergii-nosachenko) [inline src/figuresClasses.ts:24]: Create a helper function that accepts any number of args and checks them. Reuse it with all shapes.

File: src/figuresClasses.ts [hunk 6]
```diff
@@ -1,19 +1,87 @@
...
+      throw new Error('Wrong value for Triangle');
+    }
+
+    if (
+      a + b <= c
+      || a + c <= b
+      || b + c <= a
+    ) {
+      throw new Error('Wrong value for Triangle');
```
Comment (sergii-nosachenko) [inline src/figuresClasses.ts:33]: What value is wrong? You can give the more descriptive message: `Impossible to build a triangle with given sides`

File: src/figuresClasses.ts [hunk 7]
```diff
@@ -1,19 +1,87 @@
...
+    }
+  }
 
+  getArea(): number {
+    const halfPerimeter = (this.a + this.b + this.c) / 2;
+    const square = Math.sqrt(
```
Comment (sergii-nosachenko) [inline src/figuresClasses.ts:39]:
```suggestion
    const area = Math.sqrt(
```

File: src/figuresClasses.ts [hunk 8]
```diff
@@ -1,19 +1,87 @@
...
+      halfPerimeter
+       * ((halfPerimeter - this.a)
+       * (halfPerimeter - this.b)
+       * (halfPerimeter - this.c)),
+    );
+
+    return Math.floor(square * 100) / 100;
```
Comment (sergii-nosachenko) [inline src/figuresClasses.ts:46]: Create a helper function for rounding numbers.

**Review by sergii-nosachenko — CHANGES_REQUESTED** [review]

File: src/figuresClasses.ts [hunk 9]
```diff
@@ -1,19 +1,103 @@
...
+    return true;
+  }
+
+  if (sides.length === 3) {
+    const [a, b, c] = sides;
+
+    if (a + b <= c || a + c <= b || b + c <= a) {
+      return true;
+    }
+  }
```
Comment (OleksiiBlezniuk) [inline src/figuresClasses.ts:34]: Looks like separate logic and better to move it out this function

File: src/figuresClasses.ts [hunk 10]
```diff
@@ -1,19 +1,103 @@
...
+  Rectangle = 'rectangle',
+}
+
 export interface Figure {
+  color: Color;
+  shape: FigureType;
+  getArea(): number;
+}
 
+function RoundingToHundredths(number: number): number {
```
Comment (OleksiiBlezniuk) [inline src/figuresClasses.ts:19]:
```suggestion
function roundingToHundredths(number: number): number {
```

**Review by OleksiiBlezniuk — APPROVED** [review]

**Pull request closed without merging**